violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/EditMenu.java

```diff
@@ -231,8 +237,73 @@ public void actionPerformed(ActionEvent event) {
             }
         });
         this.add(selectPrevious);
-
+        
+		magnifier.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent event) {
+				SwingUtilities.invokeLater(new Runnable() {
+					@Override
+					public void run() {
+						if (isMagnifierOpened()) {
```

Comment:
brackets need to be compatible with the rest of the file.
Both if and else should start within new line.